src/main/java/org/sonatype/repository/chef/internal/util/CookBookApiAbsoluteUrlRemover.java

```diff
@@ -0,0 +1,236 @@
+import java.util.UUID;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.sonatype.goodies.common.ComponentSupport;
+import org.sonatype.nexus.repository.view.Content;
+import org.sonatype.nexus.repository.view.ContentTypes;
+import org.sonatype.nexus.repository.view.payloads.StreamPayload.InputStreamSupplier;
+import org.sonatype.repository.chef.internal.AssetKind;
+
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonToken;
+import com.google.gson.stream.JsonWriter;
+import org.apache.http.client.utils.URIBuilder;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * @since 0.0.1
+ */
+@Named
+@Singleton
+public class CookBookApiAbsoluteUrlRemover
+    extends ComponentSupport
+{
+  private ChefDataAccess chefDataAccess;
+
+  @Inject
+  public CookBookApiAbsoluteUrlRemover(final ChefDataAccess chefDataAccess) {
+    this.chefDataAccess = checkNotNull(chefDataAccess);
+  }
+
+  public Content maybeRewriteCookbookApiResponseAbsoluteUrls(final Content content, final AssetKind assetKind) throws IOException, URISyntaxException {
+    String filePath = UUID.randomUUID().toString();
+    FileOutputStream file = new FileOutputStream(filePath);
+
+    JsonReader reader = new JsonReader(new InputStreamReader(new BufferedInputStream(content.openInputStream())));
+    JsonWriter writer = new JsonWriter(new OutputStreamWriter(file, "UTF-8"));
+
+    switch (assetKind) {
+      case COOKBOOKS_LIST:
+        CookbookListJsonStreamer listStreamer = new CookbookListJsonStreamer(reader, writer);
+        listStreamer.parseJson();
+        break;
+      case COOKBOOK_DETAILS:
+        CookbookDetailsJsonStreamer detailsStreamer = new CookbookDetailsJsonStreamer(reader, writer);
+        detailsStreamer.parseJson();
+        break;
+      case COOKBOOK_DETAIL_VERSION:
+        CookbookDetailsByVersionJsonStreamer detailsByVersionStreamer = new CookbookDetailsByVersionJsonStreamer(reader, writer);
+        detailsByVersionStreamer.parseJson();
+        break;
+      case COOKBOOKS_SEARCH:
+        CookbooksSearchJsonStreamer cookbooksSearchJsonStreamer = new CookbooksSearchJsonStreamer(reader, writer);
+        cookbooksSearchJsonStreamer.parseJson();
+        break;
+    }
+
+    reader.close();
+    writer.close();
+
+    InputStream is = new FileInputStream(filePath);
+
+    File tempFile = new File(filePath);
+
+    long length = tempFile.length();
+
+    tempFile.delete();
+
+    return chefDataAccess.toContent(
+        new FileInputStreamSupplier(is),
+        length,
+        ContentTypes.APPLICATION_JSON
+    );
+  }
+
+  private class FileInputStreamSupplier
+    implements InputStreamSupplier
+  {
+    private InputStream is;
+
+    public FileInputStreamSupplier(final InputStream is) {
+      this.is = is;
+    }
+
+    @Override
+    public InputStream get() {
+      return this.is;
+    }
+  }
+
+  private class CookbooksSearchJsonStreamer
```

Comment:
You might want to consider extracting this private class out to make your testing life easier and .... wait for it .... test it ;)

Reply:
Haha

Reply:
I FINALLY TESTED IT